stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -1722,6 +1726,17 @@ impl NakamotoChainState {
         canonical_sortition_tip: &SortitionId,
         dispatcher_opt: Option<&'a T>,
     ) -> Result<Option<StacksEpochReceipt>, ChainstateError> {
+        #[cfg(any(test, feature = "testing"))]
+        {
```

Comment:
Can you put the fault injection logic here into its own method? It keeps the code cleaner.

Reply:
Sure. Refactored in bef8c905d69996b18d7fe92db95bb1b2cccee85e.